ACLPManager/manager
# DI-18419 - ACLP Error handling

## Description 📝
Added error handling for the global filters and default values for chart color and type

## Changes  🔄

1. Added error handling for all the global filters, while calling an API
2. Error handling for failing api calls of dashboard by id and resources in CloudPulseDashboardComponent
3. Handled default values of color and type if the widget config doesn't have a valid input from API

## Target release date 🗓️
28-10-2024

## Preview 📷
**Include a screenshot or screen recording of the change**

| Fixes  | Fixes   |
| ------- | ------- |
|![Screenshot 2024-10-23 at 3 28 42 PM](https://github.com/user-attachments/assets/31470b0c-9137-4a2c-8420-79663b9ad3df)|![Screenshot 2024-10-23 at 3 46 04 PM](https://github.com/user-attachments/assets/9ee590b1-bc38-4028-b828-ec52d608c817)|
|![Screenshot 2024-10-23 at 3 41 39 PM](https://github.com/user-attachments/assets/850af732-8f7f-45c4-9d3d-d2865655fd5d)|![Screenshot 2024-10-23 at 3 42 20 PM](https://github.com/user-attachments/assets/17b7fe32-8065-4295-af3a-579d91014110)|
|![Screenshot 2024-10-23 at 3 43 03 PM](https://github.com/user-attachments/assets/27eb9ddd-e9da-4cff-8be7-7f87cccbca00)|![Screenshot 2024-10-23 at 3 52 38 PM](https://github.com/user-attachments/assets/ea017538-94e5-4745-866b-ec8cdcbc3a34)|

## How to test 🧪

1. Enable mock as some of the endpoints are not available in production
2. Navigate to Monitor tab
4. For seeing error in dashboard selection component, in the serverHandler.ts file , for **monitor/services/:serviceType/dashboards** return HttpResponse.error() instead of a valid response
5. Another case for dashboard selection is in the serverHandler.ts file , for ***/monitor/services** return HttpResponse.error() instead of valid response
6. The above cases will show error in dashboard selection component
7. For seeing error in region selection component, in the serverHandler.ts file for ***/regions** return HttpResponse.error() instead of valid response, you will be able to see a valid error message
8. For seeing error in resources selection component in case of linode dashboard, in the serverHandler.ts file for ***/linode/instances** return HttpResponse.error() instead of valid response, you will be able to see a valid error message
9. For seeing error in resources selection component in case of dbaas dashboard, in the serverHandler.ts file for ***/databases/instances** return HttpResponse.error() instead of valid response, you will be able to see a valid error message

## As an Author I have considered 🤔

*Check all that apply*
